plugins/lua/spectate.lua

```diff
@@ -84,7 +84,8 @@ local function load_state()
            function() config_file:write() end
 end
 
-local config, save_state = load_state()
+-- global variables to make the UI script simpler
+_ENV.config, _ENV.save_state = load_state()
```

Comment:
```suggestion
local config, save_state = load_state()

-- called by gui/spectate
function get_config_elem(name, key)
    local elem = config[name]
    if not elem then return end
    if type(elem) == 'table' then
        return elem[key]
    end
    return elem
end

```

Reply:
both `config` and `save_state` are required though :)

how about leaving them local here, but at the end of the file making them public explicitly via `_ENV.config, _ENV.save_state = config, save_state`?

Reply:
also, now a `set_config_elem` is required :)